client/src/app/components/client-gridview/client-gridview.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit, ChangeDetectorRef } from '@angular/core';
+import { User } from 'src/app/models/user.model';
+import {AfterViewInit, ViewChild} from '@angular/core';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatTableDataSource } from '@angular/material/table';
+import { ManageMembersService } from 'src/app/services/manage-members.service';
+
+@Component({
+  selector: 'app-client-gridview',
+  templateUrl: './client-gridview.component.html',
+  styleUrls: ['./client-gridview.component.css']
+})
+
+export class ClientGridviewComponent implements OnInit {
+  displayedColumns: string[] = ['First Name', 'Is Active', 'Last Name', 'Role'];
+  dataSource: MatTableDataSource<User>;
+  users: Array<User>;
+  view = 'Client Gridview';
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+
+  constructor(
+    private manageMembersService: ManageMembersService,
+    private changeDetectorRefs: ChangeDetectorRef) {
+    this.users = new Array<User>();
+
+    // Assign the data to the data source for the table to render
+    this.manageMembersService.getAllClients()
```

Comment:
I would say that you can move this block inside a function called init() or something around that because in this case, you are initializing the table by getting all the clients. Then inside your ngOnInit() method you can call this init() method so that when a component will be created it will do all of those actions you are doing it (even though what you did is working ). Also, the ngOnInit() should be the first method after the constructor, so please reorder them.

Reply:
Fixed in commit 38fa446370aa1c6aa2e3d4df0f78677c0e8076b2